File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.java
import java.util.*;
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        
        char[] arr = number.toCharArray();
        
        StringBuilder sb = new StringBuilder();
        
        int idx = 0;
        
        for (int i = 0; i < arr.length - k; i++) {
            char max = '0';
            for (int j = idx; j <= i + k; j++) {
                if (arr[j] > max) {
                    max = arr[j];
                    idx = j + 1;
                }
            }
            sb.append(Character.toString(max));
        }
         answer = sb.toString();
        return answer;
    }
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    answer = ''
    
    num = []
    
    for i in number:
        while num and k > 0 and num[-1] < i:
            num.pop()
            k -= 1
        num.append(i)
    while k > 0:
        num.pop()
        k -= 1
    
    return ''.join(num)